modules/apps/forms-and-workflow/portal-workflow/portal-workflow-task-web/src/main/java/com/liferay/portal/workflow/task/web/internal/display/context/WorkflowTaskDisplayContext.java

```diff
@@ -1037,14 +1039,42 @@ protected WorkflowTaskSearch searchTasks(boolean searchByUserRoles)
 			searchContainer.getStart(), searchContainer.getEnd(),
 			searchContainer.getOrderByComparator());
 
-		searchContainer.setResults(results);
+		setAuthorizedSearchContainerResults(results, searchContainer);
 
 		setSearchContainerEmptyResultsMessage(
 			searchContainer, searchByUserRoles, getCompleted());
 
 		return searchContainer;
 	}
 
+	protected void setAuthorizedSearchContainerResults(List<WorkflowTask> 
+		workflowTasks, WorkflowTaskSearch searchContainer){
+
+		ThemeDisplay themeDisplay =
+			_workflowTaskRequestHelper.getThemeDisplay();
+
+		List<WorkflowTask> authorizedWorkflowTasks = new ArrayList();
+
+		for (int i = 0; i < workflowTasks.size(); i++) {
```

Comment:
@daviddotzhang, there is any special reason for you use a regular **for** instead a **enhanced for**?